src/redux/actions/SignIn.action.js

```diff
@@ -0,0 +1,27 @@
+import ACTION_CONSTANTS from '../constants/constants';
+import APIS from '../../services/api';
+
+const { LOGIN_SUCCESS, LOGIN_FAILURE } = ACTION_CONSTANTS;
+const {signInApi}=APIS;
+
+export const loginSuccess = (response) => ({
+  type: LOGIN_SUCCESS,
+  response
+});
+
+export const loginFailure = (error) => ({
+  type: LOGIN_FAILURE,
+  error
+});
+
+const signInAction = (data) => async (dispatch) => {
+  try {
+    const response = await signInApi(data);
+    dispatch(loginSuccess(response));
```

Comment:
Hello @Philipotieno , You could add the username to localstorage too since accessing user profile needs the username

Reply:
Done @a-braham

Reply:
awesome job